module/VuFind/src/VuFind/Auth/Shibboleth.php

```diff
@@ -331,6 +331,31 @@ public function logout($url)
         return $url;
     }
 
+    /**
+     * Connect user authenticated by shibboleth to library card.
+     *
+     * @param \Laminas\Http\PhpEnvironment\Request $request        Request object
+     * containing account credentials.
+     * @param \VuFind\Db\Row\User                  $connectingUser Connect newly
+     * created library card to this user.
+     *
+     * @return void
+     */
+    public function connectUser($request, $connectingUser)
+    {
+        $entityId = $this->getCurrentEntityId($request);
+        $shib = $this->getConfigurationLoader()->getConfiguration($entityId);
+        $username = $shib['prefix'] . '.' . $this->getAttribute(
```

Comment:
Should probably check that getAttribute returns a valid value here before adding a library card. If not, indicate the error with a flash message or something.